spec/app/domain/authentication/validate_security_spec.rb

```diff
@@ -0,0 +1,225 @@
+      )
+    end
+    it "raises a NotDefinedInConjur error" do
+      expect { subject }.to raise_error(Authentication::NotDefinedInConjur)
+    end
+  end
+
+  context "A whitelisted, authorized webservice and unauthorized user" do
+    subject do
+      Authentication::ValidateSecurity.new(
+        role_class: no_access_role_class,
+        webservice_resource_class: full_access_resource_class
+      ).(
+        webservice: webservice('service1'),
+          account: 'my-acct',
+          user_id: 'some-user',
+          enabled_authenticators: two_authenticator_env
+      )
+    end
+
+    it "raises a NotAuthorizedInConjur error" do
+      expect { subject }.to raise_error(Authentication::NotAuthorizedInConjur
+      )
+    end
+  end
+
+  context "Two whitelisted, authorized webservices" do
+    context "and a user authorized for only one on them" do
+      let (:webservice_resource) { 'CAN ACCESS ME' }
+      let (:partial_access_role_class) do
+        role_class(user_role_for_service(webservice_resource))
+      end
+      let (:accessible_resource_class) { resource_class(webservice_resource) }
+      let (:inaccessible_resource_class) { resource_class('CANNOT ACCESS ME') }
+
+      context "when accessing the authorized one" do
+        subject do
+          Authentication::ValidateSecurity.new(
+            role_class: partial_access_role_class,
+            webservice_resource_class: accessible_resource_class
+          ).(
+            webservice: webservice('service1'),
+              account: 'my-acct',
+              user_id: 'some-user',
+              enabled_authenticators: two_authenticator_env
+          )
+        end
+
+        it "succeeds" do
+          expect { subject }.to_not raise_error
+        end
+      end
+
+      context "when accessing the blocked one" do
+        subject do
+          Authentication::ValidateSecurity.new(
+            role_class: partial_access_role_class,
+            webservice_resource_class: inaccessible_resource_class
+          ).(
+            webservice: webservice('service1'),
+              account: 'my-acct',
+              user_id: 'some-user',
+              enabled_authenticators: two_authenticator_env
+          )
+        end
+
+        it "fails" do
+          expect { subject }.to raise_error(Authentication::NotAuthorizedInConjur)
+        end
+      end
+    end
+  end
+
+  context "An ENV lacking CONJUR_AUTHENTICATORS" do
+    subject do
+      Authentication::ValidateSecurity.new(
+        role_class: full_access_role_class,
+        webservice_resource_class: full_access_resource_class
+      ).(
+        webservice: default_authenticator_mock,
+          account: 'my-acct',
+          user_id: 'some-user',
+          enabled_authenticators: blank_env
+      )
+    end
+
+    it "the default Conjur authenticator is included in whitelisted webservices" do
+      expect { subject }.to_not raise_error
+    end
+  end
+end
```

Comment:
Excellent work on this spec.  This is a _textbook_ demonstration of how good design improves UTs.

Reply:
Thanks!